### PR TITLE
Absolute constant lookup (::Foo)

### PR DESCRIPTION
**Pull request by mvidner**
This is a fix for #180.

It is built upon #184, so please see that one first.

### CODE REVIEW

**Comment by YorickPeterse** [comment]
#184 is merged, could you rebase this PR based on the current master branch?

**Comment by mvidner** [comment]
Rebased

File: lib/ruby-lint/constant_path.rb [hunk 1]
```diff
@@ -56,9 +56,10 @@ def resolve(scope)
     end
 
     ##
-    # Returns the very first segment of the constant path as an AST node.
+    # Returns the very first segment of the constant path as a pair type+name
+    # FIXME: unused now
     #
-    # @return [RubyLint::AST::Node]
+    # @return [Array(Symbol,String)]
```
Comment (YorickPeterse) [inline lib/ruby-lint/constant_path.rb]: These changes don't seem to be related to this PR. If this method is not used anymore a separate PR should just remove it.

File: lib/ruby-lint/definition/ruby_object.rb [hunk 2]
```diff
@@ -677,6 +683,12 @@ def inspect
         return %Q(#<#{self.class}:0x#{address} #{attributes.join(' ')}>)
       end
 
+      def top_scope
+        return self if type == :root
+        scope = parents.last   # the enclosing scope?
```
Comment (YorickPeterse) [inline lib/ruby-lint/definition/ruby_object.rb]: What's the reason for this comment? Right now it doesn't really convey any useful information since it's a question.

**Comment by YorickPeterse** [comment]
@mvidner For future PRs please submit yardoc changes as a separate PR as this makes changes easier to review.

File: lib/ruby-lint/definition/ruby_object.rb (hunk 2 above)
Comment (mvidner) [inline lib/ruby-lint/definition/ruby_object.rb]: You're right. I was unsure about that. Now I have added more tests.

**Comment by mvidner** [comment]
Thank you, I've updated this to separate the yardoc, and add more tests.
